main_test.go

```diff
@@ -465,3 +465,28 @@ func TestInitDaemon(t *testing.T) {
 	require.Nil(t, err)
 	assert.Error(t, handleCLIOptions(runOpts, &installer.UBootEnv{}, dualRootfs, &menderConfig{}))
 }
+
+// Tests that the client will boot with an error message in the case of an invalid server certificate.
+func TestInvalidServerCertificateBoot(t *testing.T) {
+	tdir, err := ioutil.TempDir("", "invalidcert-test")
+	require.Nil(t, err)
+
+	logBuf := bytes.NewBuffer(nil)
+	oldLog := log.Log
+	log.Log = log.New()
```

Comment:
Replaced the whole thing with a brand new logger, and it fixed it, believe it or not...

Reply:
:+1: